Src/Support/Google.Apis/Services/BaseClientService.cs

```diff
@@ -95,10 +95,22 @@ public class Initializer
             /// <summary>Gets or sets the API Key. Default value is <c>null</c>.</summary>
             public string ApiKey { get; set; }
 
+            private string _applicationName;
             /// <summary>
             /// Gets or sets Application name to be used in the User-Agent header. Default value is <c>null</c>. 
             /// </summary>
-            public string ApplicationName { get; set; }
+            public string ApplicationName
+            {
+                get => _applicationName;
+                set
+                {
+                    if (value != null && !ProductInfoHeaderValue.TryParse(value, out var _))
```

Comment:
Is ProductInfoHeaderValue a class that should have been included in this PR?
One way this changes behaviour: if previous code set ApplicationName to something invalid, then changed it to something valid again, it would be fine, whereas now it would throw. Maybe we should only validate it when the request is made, in the name of back-compat?